homeassistant/components/leviosa_shades/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""The Leviosa shades Zone integration."""
+import logging
+
+from leviosapy import LeviosaZoneHub as tZoneHub
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.COVER]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Leviosa shades Zone from a config entry."""
+    hub_name = entry.title
+    hub_ip = entry.data["host"]
+    hub = tZoneHub(
+        hub_ip=hub_ip, hub_name=hub_name, websession=async_get_clientsession(hass)
+    )
+    await hub.getHubInfo()  # Check all is good
```

Comment:
Here you would need to catch the exceptions as well and raise `ConfigEntryError` or `ConfigEntryNotReady`.

Reply:
Suggestion accepted, thanks.